.github/workflows/pip-release.yml

```
@@ -13,6 +13,8 @@ on:
     types:
       - "published"
   workflow_dispatch:
+  push:
+  pull_request:
```

```suggestion
  push:
    branches:
      - main
  pull_request:
```

Otherwise, this is triggered twice on push and on update of pull request